sys/arm64/arm64/pmap.c

```diff
@@ -5485,9 +5514,7 @@ pmap_remove_write(vm_page_t m)
 			if (!atomic_fcmpset_long(pte, &oldpte,
 			    (oldpte | ATTR_S1_AP_RW_BIT) & ~ATTR_SW_DBM))
 				goto retry;
-			if ((oldpte & ATTR_S1_AP_RW_BIT) ==
-			    ATTR_S1_AP(ATTR_S1_AP_RW))
-				vm_page_dirty(m);
```

Comment:
Note to self/other reviewers: this didn't previously check DBM but pmap_page_dirty does (pmap_pte_dirty, called by pmap_pte_dirty, for S1 mappings, is checking DBM+RW). This is because this code was previously micro-optimised due to already being in the DBM != 0 case, i.e. manually inlined pmap_pte_dirty and removed the redundant condition. So this change is fine (and hopefully the compiler will inline and optimise some parts accordingly).